source/AI.cpp

```diff
@@ -2856,6 +2856,25 @@ Point AI::TargetAim(const Ship &ship, const Body &target)
 
 
 
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint)
+{
+	shared_ptr<const Body> shipTarget = ship.GetTargetShip();
+	shared_ptr<const Body> targetAsteroid = ship.GetTargetAsteroid();
+	shared_ptr<const Body> target = shipTarget ?: targetAsteroid;
```

Comment:
```suggestion
	shared_ptr<const Body> target = shipTarget ? shipTarget : targetAsteroid;
```
I'm surprised it even worked but we still are doing it this way everywhere else so best to continue until its changed in another PR, or not

Reply:
It’s part of the language.

The current code is short hand for the suggestion.

https://gcc.gnu.org/onlinedocs/gcc-5.3.0/gcc/Conditionals.html#Conditionals
https://stackoverflow.com/questions/1785426/c-sharp-null-coalescing-operator-equivalent-for-c
http://www.dev-tricks.net/the-operator-aka-null-coalescing-operator/

Reply:
It's a GCC extension. Let's not use language extensions because they're not exactly portable.